flowpm/tfbackground.py

```diff
@@ -496,6 +493,47 @@ def angular_diameter_distance(cosmo, a):
   return a * transverse_comoving_distance(cosmo, a)
 
 
+@tf.function
+def rk4_integration(dydt, a, y0, **kwcosmo):
+  """Runge-Kutta order 4 integration to solve ODE
+
+  Parameters
+  ----------
+  dydt: Function 
+    Takes time step t and function value y in as first two parameters and returns dydt
+
+  a: array_like or tf.TensorArray
+    Array with time stamps to evalualte the ODE at
+  
+  y0: Tensor
+    Initial value/conditions of the ODE
+
+  """
+  da = a[1:] - a[:-1]
+  steps = len(da)
+  y = tf.TensorArray(size=steps + 1, dtype=tf.float32, dynamic_size=False)
+  y = y.write(0, y0)
+
+  def body(i, yt, y):
+    t = a[i]
+    dt = da[i]
+    f1 = dydt(t, yt, **kwcosmo)
```

Comment:
ah... well.... is that going to be properly diffable? @modichirag In many cases, any arguments you want to backprop over need to be tracked as inputs of the body function. Did you check if we were getting the correct gradients?